baseplate/sidecars/live_data_watcher.py

```diff
@@ -32,6 +37,64 @@ def __init__(self, dest: str, owner: int, group: int, mode: int):
         self.group = group
         self.mode = mode
 
+    @staticmethod
+    def get_encrypted_data_from_s3(
+        bucket_name: str, file_key: str, region_name: str, sse_key: str
+    ) -> Optional[bytes]:
+        s3_client = boto3.client(
+            "s3",
+            region_name=region_name,
+        )
+        data = None
+        # The S3 data must be Server Side Encrypted and we should have the decryption key.
+        kwargs = {
+            "Bucket": bucket_name,
+            "Key": file_key,
+            "SSECustomerKey": sse_key,
+            "SSECustomerAlgorithm": "AES256",
+        }
+        try:
+            s3_object = s3_client.get_object(**kwargs)
+            # Returns bytes.
+            data = s3_object["Body"].read()
+        except ClientError as error:
+            logger.exception(
+                "Failed to load from S3. Received error code %s: %s",
+                error.response["Error"]["Code"],
+                error.response["Error"]["Message"],
+            )
+            return None
+        except ValueError as error:
+            logger.exception(error)
+            return None
+        return data
+
+    @staticmethod
+    def get_data_to_write(json_data: dict, data: Optional[bytes]) -> Optional[bytes]:
+        # Check if we have a JSON in a special format containing:
+        # data = {
+        #    "live_data_watcher_load_type": str
+        # }
+        # If the load type is 'S3', this format is an indication that we support
+        # downloading the contents of encrypted files uploaded to S3.
+        if json_data.get("live_data_watcher_load_type") == "S3":
+            try:
+                bucket_name = json_data["bucket_name"]
+                file_key = json_data["file_key"]
+                sse_key = json_data["sse_key"]
+                region_name = json_data["region_name"]
+            except KeyError:
+                # We require all of these keys to properly read from S3.
+                logger.exception(
+                    "Improper configuration found while attempting to load live data from S3."
+                )
+                return None
+            # If we have all the correct keys, attempt to read the config from S3.
+            data = NodeWatcher.get_encrypted_data_from_s3(
```

Comment:
As written, this sometimes returns None. How does the client react if it expects experiment data and finds none.

Reply:
I re-used the existing logic we have for None handing now:

[Here](https://reddit.slack.com/archives/C039HP80E21/p1649703326085289)
https://github.com/reddit/baseplate.py/blob/develop/baseplate/sidecars/live_data_watcher.py#L36

PTAL

Reply:
That logic says if the content is `None` stop watching the Node. It the behavior we want?

Reply:
This was the logic Baseplate was using before if None was passed into on_change(). Do we want to handle it differently?

I'm open to handling it differently as well. Another option is just to do nothing

Reply:
Actually, I changed it back. We no longer stop watching the Node if `data is None` in the new flow.

PTAL